StRoot/StETofCalibMaker/StETofCalibMaker.cxx

```diff
@@ -612,11 +629,41 @@ StETofCalibMaker::InitRun( Int_t runnumber )
 
         TFile* histFile = new TFile( mFileNameCalibHistograms.c_str(), "READ" );
 
-        if( !histFile || histFile->IsZombie() ) {
-            LOG_ERROR << "unable to open file: " << mFileNameCalibHistograms.c_str() << endm;
+        if( !histFile ) {
+            LOG_ERROR << "No calibration file found: " << mFileNameCalibHistograms.c_str() << endm;
             LOG_INFO  << "setting all parameters to default" << endm;
+        }else if( histFile->IsZombie() ){
+            LOG_ERROR << "Zombie calibration file: " << mFileNameCalibHistograms.c_str() << endm;
+            LOG_INFO  << "stopping execution" << endm;
+            return kStFatal;        
         }
 
+        TFile* histOffsetFile = new TFile( mFileNameOffsetHistograms.c_str(), "READ" ); //create setter!
```

Comment:
Agreeing with Rongrong. Please let us know if you want help moving that into a database table.
-Gene

Reply:
This is meant to be a local file, which is used to save temporary calibration parameters. Once those parameters are final they are uploaded to the data base and integrated into the existing time and position offset tables. This file is never needed for production running. I only pushed this to the official code because i think it should contain also all tools to create a calibration in the case that anyone else has to touch this in the future.

side remark:
This file contains run specific offsets. The way this is handled in the data base is that a timestamp for each run is created and the sum of the average offset and the run-specific offset is saved in the offset tables. I feel like there should be a more elegant way of doing this, but i don't see how for now. This should maybe be re-thought in the long run.